src/Aspire.Hosting.Azure.EventHubs/AzureEventHubsEmulatorConfigWriterHook.cs

```diff
@@ -0,0 +1,63 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Lifecycle;
+using System.Text.Json;
+
+namespace Aspire.Hosting.Azure;
+
+internal sealed class AzureEventHubsEmulatorConfigWriterHook : IDistributedApplicationLifecycleHook
+{
+    public Task AfterEndpointsAllocatedAsync(DistributedApplicationModel appModel, CancellationToken cancellationToken)
+    {
+        var eventHubsEmulatorResources = appModel.Resources.OfType<AzureEventHubsResource>().Where(x => x is { } eventHubsResource && eventHubsResource.IsEmulator);
+
+        if (!eventHubsEmulatorResources.Any())
+        {
+            // No-op if there is no Azure Event Hubs emulator resource.
+            return Task.CompletedTask;
+        }
+
+        foreach (var emulatorResource in eventHubsEmulatorResources)
+        {
+            var configFileMount = emulatorResource.Annotations.OfType<ContainerMountAnnotation>().Single(v => v.Target == AzureEventHubsEmulatorResource.EmulatorConfigJsonPath);
+
+            using var stream = new FileStream(configFileMount.Source!, FileMode.Create);
+            using var writer = new Utf8JsonWriter(stream);
+
+            writer.WriteStartObject();                      // {
+            writer.WriteStartObject("UserConfig");          //   "UserConfig": {
+            writer.WriteStartArray("NamespaceConfig");      //     "NamespaceConfig": [
+            writer.WriteStartObject();                      //       {
+            writer.WriteString("Type", "EventHub");         //         "Type": "EventHub",
+            writer.WriteString("Name", "emulatorNs1");      //         "Name": "emulatorNs1"
```

Comment:
There can be multiple hubs in this file.

@krishankumar_microsoft how should this resource be named? Or can we remove the name?

Reply:
Yeah what I'm getting at is that for each evenhub namespace in the config file ... it corresponds to a call to `AddAzureEventHubs` in the app host - that resource has a name so we should use that name here.

Reply:
I get it now, thanks.

Reply:
I will have to revert this change, 

```
Expected string to be "emulatorns1" with a length of 11 because NamespaceName is non-modifiable.Only supported emulatorns1, but "eventhubns" has a length of 10, differs near "ven" (index 1).
```

Reply:
odd

Reply:
wow

Reply:
Yea, the team says this is hardcoded for now. So proceed as you have it.